src/lib.rs

```diff
@@ -0,0 +1,19 @@
+#![no_std]
```

Comment:
```suggestion
#![cfg_attr(not(feature = "std"), no_std)]
```

Reply:
There is no need to use features here since we won't ever use `std` in the crate - only in the implementations

Reply:
why do we want this crate to be no-std?

Reply:
Its the other way around. There is no reason to make it `std`, so being `no-std` is more powerful and flexible.